pkg/blobstore/cloud_blob_access.go

```diff
@@ -59,14 +61,25 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 	}
 	// In case of an error (e.g. network failure), we cancel before closing to
 	// request the write to be aborted.
```

Comment:
Please take some time to read this comment. There is a reason why the code is the way it is: the order in which `cancel()` and `w.Close()` is called matters here. This instructs go-cloud to cancel the transmission of the current file.

Reply:
Ah, I hadn't noticed that the order flips depending on whether the close was unexpected or not, but I really should have.